service/controller/v25/adapter/guest_auto_scaling_group.go

```diff
@@ -49,8 +49,18 @@ func (a *GuestAutoScalingGroupAdapter) Adapt(cfg Config) error {
 	a.ASGMinSize = minWorkers
 	a.ASGType = key.KindWorker
 	a.ClusterID = key.ClusterID(cfg.CustomObject)
-	a.MaxBatchSize = workerCountRatio(currentDesiredMinWorkers, asgMaxBatchSizeRatio)
-	a.MinInstancesInService = workerCountRatio(currentDesiredMinWorkers, asgMinInstancesRatio)
+	a.MaxBatchSize = strconv.Itoa(workerCountRatio(currentDesiredMinWorkers, asgMaxBatchSizeRatio))
+
+	minInstancesInService := workerCountRatio(currentDesiredMinWorkers, asgMinInstancesRatio)
+	if minInstancesInService != 0 && minInstancesInService == maxWorkers {
+		// MinInstancesInService must be less than the autoscaling group's MaxSize.
+		// This should only ever be an issue if the min and max workers are exactly one.
+		// If this is the case then the cluster will most likely go down while the new worker
+		// is coming up and the old one is removed.
```

Comment:
This comment makes me think if we can't make the condition stronger and check if min and max workers are qual to `1`.

Reply:
I changed this condition PTAL